### PR TITLE
Updates docs structure to prepare for Github Pages

### PR DESCRIPTION
**Pull request by adamnova**
### Details on the issue fix or feature implementation

Restructures docs a bit to improve readability in Github Pages while preserving the ability to browse them in Github.

### Confirm the following

- [x]  I started this PR by branching from the head of the default branch
- [x]  I have targeted the PR to merge into the default branch
- [x]  I have included unit tests for the issue/feature
- [x]  I have successfully run a local build

### CODE REVIEW

File: docs/README.md
Comment (martintmk) [inline docs/README.md:1]: The convention is to use README.md for readme files. Does it causes any issues with focfx?

File: docs/readme.md [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+|[Timeout](strategies/timeout.md)|No|Beyond a certain wait, a success result is unlikely.| *Don't wait forever*  |Guarantees the caller won't have to wait beyond the timeout. |
+|[Rate Limiter](strategies/rate-limiter.md)|No|Limiting the rate a system handles requests is another way to control load. <br/><br/> This can apply to the way your system accepts incoming calls, and/or to the way you call downstream services. | *Slow down a bit, will you?*  |Constrains executions to not exceed a certain rate. |
+|[Fallback](strategies/fallback.md)|Yes|Things will still fail - plan what you will do when that happens.| *Degrade gracefully*  |Defines an alternative value to be returned (or action to be executed) on failure. |
+|[Hedging](strategies/hedging.md)|Yes|Things can be slow sometimes, plan what you will do when that happens.| *Hedge your bets*  | Executes parallel actions when things are slow and waits for the fastest one.  |
+
+Visit the [resilience strategies](strategies/readme.md) section to understand their structure and explore various configuration methods.
+
+## Articles
```
Comment (martintmk) [inline docs/readme.md:22]: Add `Next steps` section that will point to quick start

File: docs/README.md
Comment (adamnova) [reply docs/README.md:1]: This was deliberate change as github pages are case sensitive and I considered lowercase readme.md in docs the better alternative compared to using /README.md in Github Pages inside the url, which could cause a lot more confusion.

Comment (martintmk) [reply docs/README.md:1]: let's stick with GitHub convention and then we can do post-processing when publishing the docs.

**Pull request merged**